src/platform/windows/misc.cpp

```diff
@@ -484,44 +693,84 @@ namespace platf {
         &process_info);
+    });  // Use the results of the launch to create a bp::child object
+    return create_boost_child_from_results(ret, cmd, ec, process_info, group);
+  }
 
-      if (!ret) {
-        auto error = GetLastError();
-
-        if (error == 740) {
-          BOOST_LOG(info) << "Could not execute previous command because it required elevation. Running the command again with elevation, for security reasons this will prompt user interaction."sv;
-          startup_info.StartupInfo.wShowWindow = SW_HIDE;
-          startup_info.StartupInfo.dwFlags = startup_info.StartupInfo.dwFlags | STARTF_USESHOWWINDOW;
-          std::wstring elevated_command = L"tools\\elevator.exe ";
-          elevated_command += wcmd;
-
-          // For security reasons, Windows enforces that an application can have only one "interactive thread" responsible for processing user input and managing the user interface (UI).
-          // Since UAC prompts are interactive, we cannot have a UAC prompt while Sunshine is already running because it would block the thread.
-          // To work around this issue, we will launch a separate process that will elevate the command, which will prompt the user to confirm the elevation.
-          // This is our intended behavior: to require interaction before elevating the command.
-          ret = CreateProcessAsUserW(shell_token,
-            nullptr,
-            (LPWSTR) elevated_command.c_str(),
-            nullptr,
-            nullptr,
-            !!(startup_info.StartupInfo.dwFlags & STARTF_USESTDHANDLES),
-            EXTENDED_STARTUPINFO_PRESENT | CREATE_UNICODE_ENVIRONMENT | CREATE_NEW_CONSOLE | CREATE_BREAKAWAY_FROM_JOB,
-            env_block.data(),
-            start_dir.empty() ? nullptr : start_dir.c_str(),
-            (LPSTARTUPINFOW) &startup_info,
-            &process_info);
-        }
-      }
+  /**
+ * @brief Runs a command on the users profile, without elevation
+ *
+ * This function launches a child process as an unprivileged user, using the current user's environment
+ * and a specific working directory. If the launch is successful, a `bp::child` object representing the new
+ * process is returned. Otherwise, an error code is returned.
+ *
+ * @param cmd The command to run
+ * @param working_dir The working directory for the new process
+ * @param env The environment variables to use for the new process
+ * @param file A file object to redirect the child process's output to (may be nullptr)
+ * @param ec An error code, set to indicate any errors that occur during the launch process
+ * @param group A pointer to a `bp::group` object to which the new process should belong (may be nullptr)
+ *
+ * @return A `bp::child` object representing the new process, or an empty `bp::child` object if the launch fails
+ */
+  bp::child
+  run_unprivileged(const std::string &cmd, boost::filesystem::path &working_dir, bp::environment &env, FILE *file, std::error_code &ec, bp::group *group) {
```

Comment:
It looks like `run_unprivileged()` and `run_privileged()` are pretty similar. Rather than separating out the parts like `create_startup_info()` and `create_boost_child_from_results()`, I think you can consolidate them into a single run function that just takes a parameter of whether to elevate or not like you did with `run_command()`.